File: docs/v0.5-fr/02-components/snippet-example.js
/**
 * Un fichier que la page Snippet lit, au lieu de le recopier.
 *
 * Il est installé avec la documentation : les exemples de cette page
 * fonctionnent donc dans votre projet comme ils fonctionnent sur ce site.
 */

export const GREETINGS = { en: 'Hello', fr: 'Bonjour' };

export function greet(name, lang = 'en') {
  // #region guard
  if (!name) {
    throw new Error('greet needs a name');
  }
  // #endregion

  return `${GREETINGS[lang] ?? GREETINGS.en}, ${name}!`;
}

File: docs/v0.5-fr/02-components/snippet-example.py
"""Un second fichier, dans un autre langage, pour que la page montre ce qui change.

Rien ici n'est lu par l'outil : c'est seulement la source d'un exemple.
L'extension est ce qui dit au bloc quel langage il contient, et donc quelle
marque et quelle couleur il porte.
"""

SKIPPED = {"le", "la", "les", "de", "des", "du"}


def slug_of(title: str) -> str:
    """Transforme le titre d'une page en dernier segment de son adresse."""
    # #region guard
    if not isinstance(title, str) or not title.strip():
        raise TypeError("slug_of needs a title")
    # #endregion

    words = "".join(c if c.isalnum() else " " for c in title.lower()).split()
    return "-".join(w for w in words if w not in SKIPPED)
